bliss.js

```diff
@@ -483,57 +507,43 @@ $.setProps = {
 		}
 	},
 
-	once: function(val) {
-		if (arguments.length == 2) {
-			val = {};
-			val[arguments[0]] = arguments[1];
-		}
-
+	once: overload(function(events, callback){
+		events = events.split(/\s+/);
 		var me = this;
+		var once = function() {
+			events.forEach(function(event){
+				me.removeEventListener(event, once);
+			});
 
-		$.each(val, function(events, callback){
-			events = events.split(/\s+/);
-
-			var once = function() {
-				events.forEach(function(event){
-					me.removeEventListener(event, once);
-				});
-
-				return callback.apply(me, arguments);
-			};
+			return callback.apply(me, arguments);
+		};
 
-			events.forEach(function (event) {
-				me.addEventListener(event, once);
-			});
+		events.forEach(function (event) {
+			me.addEventListener(event, once);
 		});
-	},
+	}, 0),
 
 	// Event delegation
-	delegate: function(val) {
-		if (arguments.length === 3) {
-			// Called with ("type", "selector", callback)
-			val = {};
-			val[arguments[0]] = {};
-			val[arguments[0]][arguments[1]] = arguments[2];
-		}
-		else if (arguments.length === 2) {
-			// Called with ("type", selectors & callbacks)
-			val = {};
-			val[arguments[0]] = arguments[1];
-		}
-
-		var element = this;
-
-		$.each(val, function (type, callbacks) {
-			element.addEventListener(type, function(evt) {
-				for (var selector in callbacks) {
-					if (evt.target.closest(selector)) {
-						callbacks[selector].call(this, evt);
-					}
+	delegate: overload(function (type, selector, callback) {
```

Comment:
I’m very confused. What do you mean the function needs to be called only once? Why can't it just add multiple event listeners? It looks to me as all that needs to be done is to remove the loop there. I'm going to push a commit soon, please take a look and tell me what I'm missing here.